task_processing/plugins/mesos/constraints.py

```diff
@@ -0,0 +1,89 @@
+import abc
+import re
+
+from pyrsistent import field
+from pyrsistent import PRecord
+
+
+class Operator(object):
+
+    @abc.abstractmethod
+    def __call__(self, arg1, arg2):
+        raise NotImplementedError('__call__() method is not implemented')
+
+
+class EqualsOperator(Operator):
+
+    def __call__(self, expected_value, actual_value):
+        return expected_value == actual_value
+
+
+class NotEqualsOperator(Operator):
+
+    def __call__(self, expected_value, actual_value):
+        return expected_value != actual_value
+
+
+class LikeOperator(Operator):
+
+    def __call__(self, like_pattern, actual_value):
+        escaped_pattern = re.escape(like_pattern)
+        re_pattern = escaped_pattern.replace('\\%', '.*').replace('_', '.')
+        return re.match(re_pattern, actual_value)
+
+
+class UnlikeOperator(LikeOperator):
+
+    def __call__(self, like_pattern, actual_value):
+        return not LikeOperator.__call__(self, like_pattern, actual_value)
+
+
+OPERATORS = {
+    'EQUALS': EqualsOperator(),
+    '==': EqualsOperator(),
+    'NOTEQUALS': NotEqualsOperator(),
+    '!=': NotEqualsOperator(),
+    'LIKE': LikeOperator(),
+    'UNLIKE': UnlikeOperator(),
+}
+
+
+def _dict_matches_constraint(kv, constraint):
+    c = Constraint(
+        attribute=constraint[0],
+        operator=constraint[1],
+        value=constraint[2]
+    )
+    actual_value = kv.get(c.attribute)
+    # If the dictionary doesn't contain an attribute from the constraint then
+    # the constraint is satisfied.
+    if actual_value is None:
+        return True
+
+    # The operator names have already been validated by the validator in
+    # `MesosTaskConfig`, so it's guaranteed that it's in `OPERATORS`.
+    return OPERATORS[c.operator](c.value, actual_value)
+
+
+def dict_matches_constraints(kv, constraints):
```

Comment:
can you call this 'attributes_match_all_constraints'?

Reply:
yes, and can you rename kv => attributes?